multiraft/multiraft.go

```diff
@@ -644,10 +644,7 @@ func (s *state) start() {
 	})
 }
 
-func (s *state) removePending(g *group, prop *proposal, err error) {
-	if prop == nil {
-		return
-	}
+func (s *state) removePending(g *group, prop proposal, err error) {
```

Comment:
This method mutates `prop`, so this one needs to stay a pointer, as does the `pending` map.